core/src/main/java/com/novoda/noplayer/internal/exoplayer/drm/ExoPlayerDrmSessionCreator.java

```diff
@@ -0,0 +1,90 @@
+package com.novoda.noplayer.internal.exoplayer.drm;
+
+import android.os.Handler;
+import android.util.Pair;
+
+import com.google.android.exoplayer2.drm.DefaultDrmSessionEventListener;
+import com.google.android.exoplayer2.drm.DefaultDrmSessionManager;
+import com.google.android.exoplayer2.drm.DrmSession;
+import com.google.android.exoplayer2.drm.FrameworkMediaCrypto;
+import com.google.android.exoplayer2.drm.FrameworkMediaDrm;
+import com.google.android.exoplayer2.drm.OfflineLicenseHelper;
+import com.google.android.exoplayer2.drm.UnsupportedDrmException;
+import com.novoda.noplayer.drm.KeyRequestExecutor;
+import com.novoda.noplayer.internal.drm.provision.ProvisionExecutor;
+import com.novoda.noplayer.model.KeySetId;
+
+import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+
+import androidx.annotation.Nullable;
+
+class ExoPlayerDrmSessionCreator implements DrmSessionCreator {
```

Comment:
This is the important class which created a unified `DrmSessionManager` for downloads and streaming based on the `KeySetId`.